src/components/ui/cart/cart.lineitem.jsx

```diff
@@ -0,0 +1,187 @@
+                    {
+                        method: 'put',
+                        headers: {
+                            'Content-Type': 'application/json',
+                            'Authorization': localStorage.getItem(Config.cortexApi.scope + '_oAuthToken')
+                        },
+                        body: JSON.stringify({
+                            quantity: this.state.quantity
+                        })
+                    })
+                    .then(() => {
+                        this.props.handleQuantityChange()
+                    })
+                    .catch(error => {
+                        console.log(error)
+                    });
+            });
+        });
+    }
+    handleRemoveBtnClicked() {
+        login().then(() => {
+            fetch(this.props.item.self.href,
+                {
+                    method: 'delete',
+                    headers: {
+                        'Content-Type': 'application/json',
+                        'Authorization': localStorage.getItem(Config.cortexApi.scope + '_oAuthToken')
+                    }
+                })
+                .then(() => {
+                    this.props.handleQuantityChange()
+                })
+                .catch(error => {
+                    console.log(error);
+                })
+        });
+    }
+    renderUnitPrice() {
+        var listPrice = this.props.item['_price'][0]['list-price'][0]['display'];
+        var purchasePrice = this.props.item['_price'][0]['purchase-price'][0]['display'];
+        if (listPrice !== purchasePrice) {
+            return (
+                <ul className="cart-lineitem-price-container">
+                    <li className="cart-unit-list-price" data-region="itemListPriceRegion">{listPrice}</li>
+                    <li className="cart-unit-purchase-price">{purchasePrice}</li>
+                </ul>
+            );
+        } else {
+            return (
+                <ul className="cart-lineitem-price-container">
+                    <li className="cart-unit-purchase-price">{purchasePrice}</li>
+                </ul>
+            );
+        }
+    }
+    renderOptions() {
+        var options = this.props.item['_item'][0]['_definition'][0]['_options'];
+        if (options) {
+            return (
+                options[0]['_element'].map((option) => {
+                    return (
+                        <li className="cart-lineitem-option" key={'_' + Math.random().toString(36).substr(2, 9)}>
+                            <label className="cart-lineitem-option-name">{option['display-name']}:&nbsp;</label>
+                            <span className="cart-lineitem-option-value">{option['_value'][0]['display-name']}</span>
+                        </li>
+                    );
+                })
+            );
+        }
+    }
+    render() {
+        var stock = this.props.item['_availability'][0]['state'] === "AVAILABLE" ? "In Stock" : "Out of Stock";
```

Comment:
You could (if have time later) go as far as returning the span for the icon here and only show it if the item is available 🙂